tests/variable/variable.cpp

```diff
@@ -1699,3 +1699,38 @@ TEST(Variable, variableInterfaceDontDowngrade)
     EXPECT_EQ("", v3->interfaceType());
     EXPECT_EQ("public", v4->interfaceType());
 }
+
+TEST(Variable, addEquivalenceToImportedVariable)
+{
+    // This won't work ... but it's what I'd want to be able to do ...
+    auto model = libcellml::Model::create("importing_model");
+
+    // Set up the local component
+    auto localComponent = libcellml::Component::create("localComponent");
+    auto x = libcellml::Variable::create("x");
+    x->setUnits("dimensionless");
+    x->setInterfaceType("public_and_private");
+
+    // Set up the imported component
+    auto importedComponent = libcellml::Component::create("importedComponent");
+    model->addComponent(importedComponent);
+
+    auto importer = libcellml::ImportSource::create();
+    importer->setUrl("importedModel.cellml");
+    importedComponent->setImportSource(importer);
+    importedComponent->setImportReference("importMe");
+
+    // Create equivalent variables: THIS BIT I DON'T KNOW HOW TO DO!
+    // The importedComponent is not instantiated, so can't point to any of its variables...
+    // libcellml::Variable::addEquivalence(x, importedComponent->variable("x"));
+    libcellml::Variable::addEquivalence(localComponent, "x", importedComponent, "x");
```

Comment:
Why aren't you using the reference to 'x' that you have for the local component?  If you believe that imported component also has a variable 'x' you can create that variable and add it to the component.  With that you can use the API.

I could see us having the API (component, variable, component, string)

Reply:
And what about when both components are imported? Then you'd need (component, string, component, string)

Reply:
And if I add a variable to the importedComponent, isn't that a bit weird? Then I have two variables with the same name?

Reply:
Having two variables with the same name isn't weird, there are loads of duplicate variable names.

Reply:
I think then that you would make the whole set:

- (component, variable, component, string)
- (component, string, component, string)
- (component, string, component, variable)
- (component, string, component, string)

Reply:
Duplicate variable names, sure, but not in the same component?
But yes to the whole set of combinations, though you have a duplicate in there ... :)

Reply:
or, better:
variable, variable
variable, component, string,
(possibly?) component, string, variable (?)
component, string, component, string

Reply:
I'd thought last time that I'd talked with @hsorby about this issue, the idea was that you would simply add variables to the imported component and then the equivalences work as normal. But then when printing the model the imported component would not have the variables printed (as they make no sense) and but the equivalences (connections) would still be printed.

Reply:
Yes but I can't see any reason why you couldn't add the string form for convenience.

This is possible at the moment but you can only do it one way. I think if adding a string form would make it easier for users and doesn't make problems and tests still pass at the moment I cannot say it is wrong to add the additional API.

Reply:
@nickerso I don't quite follow the logic of adding the variable to the component when it's really already there in the import?  Do you also have to add units for the model (before flattening) to pass validation?  Do you have to also have the units in your instantiated model too?  I'm missing something ...

Reply:
> Yes but I can't see any reason why you couldn't add the string form for convenience.

Yep, totally agree.

Reply:
> it's really already there in the import

Its only there once the import is resolved and instantiated. For exactly this case of handling CellML connections as variable equivalences it is sometimes necessary to make these dummy variables. They are never used for anything else (I think) so units/interfaces/etc. are not needed.

Reply:
and since this would be an imported component these dummy variables are never used in flattening, validation, printing...

Reply:
The units are used in validation because we now test for unit matching across equivalent variables.  :( I get an error:
```Validator error[0]:
     Description: Variable 'g_Na' in component 'sodiumChannel' has units of 'mS_per_cm2' and an equivalent variable 'g_Na' in component 'sodiumChannelInit' with non-matching units of ''. The mismatch is: 
    See section X.Y.Z in the CellML specification.```

... where `sodiumChannel` is a local component, and `sodiumChannelInit` is an import.  Is this expected? Or should the validator not test for units in a component which has a non-null import source?

Reply:
Also interface errors from the dummy variables: 
```
Description: Variable 'm' in component 'mGateInit' has no interface type set. The interface type required is 'public'.
```